UsingSeal.md

```diff
@@ -191,28 +191,38 @@ Currently, access to above key servers is permissive. However, rate limiting is
 
 ## For key server operators
 
-### Setup
-Use the `seal-cli` tool to generate a new master key using `cargo run --bin seal-cli genkey`.
+The key server can run either in `Open` mode or `Permissioned`.
+`Open` mode allows querying keys for any onchain package and uses a single IBE master key for all of them.
+`Permissioned` mode restricts queries to manually approved packages, associated with a specific client. It also supports importing and exporting a key for the specific client if needed. More on both options in the following.
 
-Key servers can be registered onchain to enable discoverability. To register a key server, call the `create_and_transfer_v1` function in the `seal::key_server` module. For example:
+### Open mode
+In this mode Seal policies of any package can be queried.
+This is an ideal option for testing Seal, or if the key server does not have any direct liabilty to users (*best-effort* service).
+
+First, a master BLS keypair should be generated.
+Use the `seal-cli` tool for that using `cargo run --bin seal-cli genkey`.
+It outputs the master secret key and the public key.
+
+Key servers should be registered onchain to enable discoverability.
+To register a key server, call the `create_and_transfer_v1` function in the `seal::key_server` module. For example:
 
 ```shell
 sui client call --function create_and_transfer_v1 --module key_server --package 0x62c79dfeb0a2ca8c308a56bde530ccf3846535e1623949d45c90d23128afff52 --args mysten-dev-1 https://seal-key-server-testnet-1.mystenlabs.com 0 0xa023acbf600401017ee17bf918106ea9911914ca017aa3ab9ab5c64beb9bb5236fd9d4d5b5645dc3bc0d4f732ed04fc60d14b9f37987fe5eeb4db07fc0982904ce1ed0b07607ae2e99086e141f6c6a1df6def5f5d434ca7c09856a3750c92969 --gas-budget 10000000
 ```
 
-Run the server using `cargo run --bin key-server` either with the environment variable `CONFIG_PATH` pointing to a .yaml configuration file (see [crates/key-server/key-server-config.yaml] for an example) or by setting the required environment variables directly:
-- `MASTER_KEY` is the master secret key generated by the `seal-cli` tool.
-- `KEY_SERVER_OBJECT_ID` is the object id of the registered key server.
-- `NETWORK` specifies the network to connect to, such as `testnet`, `mainnet`, or other supported environments. To use a custom full node, set `NETWORK` to `custom`. When using the `custom` option, you must also set the `NODE_URL` environment variable with the URL of your Full node endpoint.
+Run the server using `cargo run --bin key-server` with the environment variable 
```

Comment:
For this entire statement before the example shell commands, here's a suggested edit:

> To start the key server in open mode, run the command `cargo run --bin key-server`, but before running the server, set the following environment variables:
> 
> - MASTER_KEY: The master secret key generated using the `seal-cli genkey` command.
> - CONFIG_PATH: The path to a .yaml configuration file that specifies key server settings. For the configuration file format, see the [example config](crates/key-server/key-server-config.yaml).
> 
> In the config file, make sure to:
> - Set the mode to `Open`.
> - Set the `key_server_object_id` field to the ID of the key server object you registered on-chain.